File: gm-utils/src/main/java/src/commom/utils/string/StringBox.java
package src.commom.utils.string;

import src.commom.utils.object.Null;
import src.commom.utils.object.ObjJs;
import src.commom.utils.object.UJson;

public class StringBox extends ObjJs {

	private String value = "";

	public StringBox(String s) {
		set(s);
	}
	
	public String get() {
		return value;
	}

	public String set(String s) {
		if (Null.is(s)) {
			value = "";
		} else {
			value = s;
		}
		return value;
	}

	public StringBox add(String s) {
		set(get()+s);
		return this;
	}

	public boolean isEmpty() {
		return StringEmpty.is(value);
	}

	public String removeLeft(int count) {
		String x = value.substring(0, count);
		value = value.substring(count);
		return x;
	}

	public String removeRight(int count) {
		String x = StringRight.get(value, count);
		value = StringRight.ignore(value, count);
		return x;
	}

	public StringBox replace(String a, String b) {
		set(StringReplace.exec(value, a, b));
		return this;
	}

	public StringBox replaceWhile(String a, String b) {
		set(StringReplace.whilee(value, a, b));
		return this;
	}
	
	public void clear() {
		set("");
	}

	@Override
	public String toString() {
		return value;
	}

	public boolean eq(String s) {
		return StringCompare.eq(get(), s);
	}
	@Override
	protected String toJsonImpl() {
		return "{" + UJson.itemString("value", value) + "}";
	}
}

File: gm-utils/js/src/commom/utils/string/StringBox.js
import Null from '../object/Null';
import ObjJs from '../object/ObjJs';
import StringCompare from './StringCompare';
import StringEmpty from './StringEmpty';
import StringReplace from './StringReplace';
import StringRight from './StringRight';
import UJson from '../object/UJson';

export default class StringBox extends ObjJs {

	constructor(s) {
		super();
		this.set(s);
	}

	get() {
		return this.value;
	}

	set(s) {
		if (Null.is(s)) {
			this.value = "";
		} else {
			this.value = s;
		}
		return this.value;
	}

	add(s) {
		this.set(this.get()+s);
		return this;
	}

	isEmpty() {
		return StringEmpty.is(this.value);
	}

	removeLeft(count) {
		let x = this.value.substring(0, count);
		this.value = this.value.substring(count);
		return x;
	}

	removeRight(count) {
		let x = StringRight.get(this.value, count);
		this.value = StringRight.ignore(this.value, count);
		return x;
	}

	replace(a, b) {
		this.set(StringReplace.exec(this.value, a, b));
		return this;
	}

	clear() {
		this.set("");
	}

	toString() {
		return this.value;
	}

	eq(s) {
		return StringCompare.eq(this.get(), s);
	}
	toJsonImpl() {
		return "{" + UJson.itemString("value", this.value) + "}";
	}
}
